lib/RawModule.js

```diff
@@ -2,47 +2,67 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var Module = require("./Module");
-var OriginalSource = require("webpack-sources").OriginalSource;
-var RawSource = require("webpack-sources").RawSource;
-
-function RawModule(source, identifier, readableIdentifier) {
-	Module.call(this);
-	this.sourceStr = source;
-	this.identifierStr = identifier || this.sourceStr;
-	this.readableIdentifierStr = readableIdentifier || this.identifierStr;
-	this.cacheable = true;
-	this.built = false;
+"use strict";
+
+const Module = require("./Module"),
+	OriginalSource = require("webpack-sources").OriginalSource,
+	RawSource = require("webpack-sources").RawSource;
+
+module.exports = class RawModule extends Module {
+
+	constructor(source, identifier, readableIdentifier) {
+		super()
+		this.sourceStr = source;
+		this.identifierStr = identifier || this.sourceStr;
+		this.readableIdentifierStr = readableIdentifier || this.identifierStr;
+		this.cacheable = true;
+		this.built = false;
+	}
+
+	identifier() {
+		return this.identifierStr;
+	}
+
+	size() {
+		return this.sourceStr.length;
+	}
+
+	readableIdentifier(requestShortener) {
+		return requestShortener.shorten(this.readableIdentifierStr);
+	}
+
+	needRebuild() {
+		return false;
+	}
+
+	build(options, compilations, resolver, fs, callback) {
+		this.builtTime = new Date().getTime();
+		callback();
+	}
+
+	source() {
+		if(this.useSourceMap)
+			return new OriginalSource(this.sourceStr, this.identifier());
+		else
+			return new RawSource(this.sourceStr);
+	}
+
+	getSourceHash() {
```

Comment:
`getSourceHash` `getAllModules` `createTemplate` `getTemplateArguments` doesn't exist in original source.